app/src/main/java/com/allan/boardbuddies/fragments/EditBoardFragment.java

```diff
@@ -0,0 +1,116 @@
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class EditBoardFragment extends Fragment implements NavigationBarView.OnItemSelectedListener {
+    private EditBoardViewModel editBoardViewModel;
+    private TextView boardTitle;
+    private CanvasView canvasView;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_edit_board, container, false);
+        Toolbar toolbar = view.findViewById(R.id.board_toolbar);
+
+        boardTitle = view.findViewById(R.id.board_title);
+        canvasView = view.findViewById(R.id.canvas_view);
+        editBoardViewModel = new ViewModelProvider(requireActivity()).get(EditBoardViewModel.class);
+
+        ViewTreeObserver viewTreeObserver = canvasView.getViewTreeObserver();
+        if (viewTreeObserver.isAlive()) {
+            viewTreeObserver.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+                @Override
+                public void onGlobalLayout() {
+                    canvasView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                    canvasView.init(canvasView.getWidth(), canvasView.getHeight());
+                }
+            });
+        }
+
+        toolbar.setNavigationOnClickListener(v -> {
+            editBoardViewModel.saveBoard(boardTitle.getText().toString(), canvasView.getStrokes(), canvasView.getTextBoxes());
+            Navigation.findNavController(requireActivity(), R.id.nav_host).popBackStack();
+        });
+
+        editBoardViewModel.setBoard(getArguments().getInt("position", -1));
+        boardTitle.setText(editBoardViewModel.getLocalBoard().getTitle());
+        canvasView.setStrokes(new ArrayList<Stroke>(editBoardViewModel.getLocalBoard().getStrokes().stream().map(Stroke::new).collect(Collectors.toList())));
+        canvasView.setTexts(new ArrayList<TextBox>(editBoardViewModel.getLocalBoard().getTexts().stream().map(TextBox::new).collect(Collectors.toList())));
+
+        BottomNavigationView bottomNavigationView = view.findViewById(R.id.board_bottom_nav);
+        bottomNavigationView.setOnItemSelectedListener(this);
+
+        getChildFragmentManager().setFragmentResultListener(Constants.TEXT_DIALOG_FRAGMENT_ADD_TEXT_REQUEST_KEY, this, new FragmentResultListener() {
+            @Override
+            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle bundle) {
+                String result = bundle.getString(Constants.TEXT_DIALOG_FRAGMENT_ADDED_TEXT);
+                String resultKey = bundle.getString("resultKey");
+                if ("add_text".equals(resultKey) && !result.isEmpty()){
+                    canvasView.newTextBox(result);
+                } else if ("edit_canvas_title".equals(resultKey)){
+                    boardTitle.setText(result);
+                }
+            }
+        });
```

Comment:
as discussed, can we move this into the viewmodel? TextDialogFragment can just modify either the title of the board or add a new text, and it will reflect here

Reply:
not in scope, next ticket